routing/router_test.go

```diff
@@ -602,13 +639,157 @@ func TestSendPaymentErrorRepeatedFeeInsufficient(t *testing.T) {
 	if !bytes.Equal(route.Hops[0].PubKeyBytes[:],
 		ctx.aliases["phamnuwen"].SerializeCompressed()) {
 
-		t.Fatalf("route should go through satoshi as first hop, "+
+		t.Fatalf("route should go through pham nuwen as first hop, "+
 			"instead passes through: %v",
 			getAliasFromPubKey(route.Hops[0].PubKeyBytes[:],
 				ctx.aliases))
 	}
 }
 
+// TestSendPaymentErrorFeeInsufficientPrivateEdge tests that if we receive
+// a fee related error from a private channel that we're attempting to route
+// through, then we'll update the fees in the route hints and successfuly
+// route through the private channel in the second attempt.
+func TestSendPaymentErrorFeeInsufficientPrivateEdge(t *testing.T) {
+	t.Parallel()
+
+	const startingBlockHeight = 101
+	ctx, cleanUp, err := createTestCtxFromFile(startingBlockHeight, basicGraphFilePath)
```

Comment:
There are currently two ways of writing these kind of path finding tests. One is the way you did it, using the json file. The other option is used in for example `TestChannelUpdateValidation`. It allows programmatically building the test graph and generates some values like the keys by itself (as the actual value isn't important for the tests, it just needs a value). You could consider using that instead. I think it is clearer than keep adding new channels to the json file.

Reply:
I think that you will still have the problem of needing a private key even if you use TestChannelUpdateValidation approach, as the validation of the error message is part of the tested code. Anyhow, I like the approach of using the basic graph in many tests all based on a non trivial topology. It requires the writer of the test to rethink whether the logic indeed covers all scenarios, beyond the simplest topology needed for testing the particular feature or fix she was aiming to do.

Reply:
That's okay, just wanted to mention the option.